src/driver/src/DefaultDriver.cpp

```diff
@@ -163,35 +164,26 @@ DefaultDriver::OutcomeCode doRunLogic(const DefaultDriver::ProgramOptions& optio
     std::atomic<DefaultDriver::OutcomeCode> outcomeCode = DefaultDriver::OutcomeCode::kSuccess;
 
     std::mutex reporting;
-    auto threadsPtr = std::make_unique<std::vector<std::thread>>();
-    threadsPtr->reserve(workloadContext.actors().size());
-    std::transform(cbegin(workloadContext.actors()),
-                   cend(workloadContext.actors()),
-                   std::back_inserter(*threadsPtr),
-                   [&](const auto& actor) {
-                       return std::thread{[&]() {
-                           {
-                               auto ctx = startedActors.start();
-                               ctx.addDocuments(1);
-
-                               std::lock_guard<std::mutex> lk{reporting};
-                               ctx.success();
-                           }
-
-                           runActor(actor, outcomeCode, orchestrator);
-
-                           {
-                               auto ctx = finishedActors.start();
-                               ctx.addDocuments(1);
-
-                               std::lock_guard<std::mutex> lk{reporting};
-                               ctx.success();
-                           }
-                       }};
-                   });
-
-    for (auto& thread : *threadsPtr)
-        thread.join();
+    parallelRun(workloadContext.actors(),
```

Comment:
@aggrand it looks like here should have the total number of threads, or at least could compute it (workloadContext.actors().size()) could we put a for loop (or something fancier) of the form

```
std::vector<int> RNGSeeds;
for (auto actor : workloadContext.actors())
    for (int i =; i < actor.size(): i++)
        {
          RNGSeeds.push_back(workloadContext._rng());
        }
```
Or directly construct the rng registry right here. 

You can't directly access the _rng today, but that part seems easy enough to expose.

Reply:
If this was a bottleneck in itself, you could try cloning the rng, and use discard to advance the rng as needed to match the serial execution.